pkg/core/dao/dao.go

```diff
@@ -363,13 +342,17 @@ func makeStorageItemKey(prefix storage.KeyPrefix, id int32, key []byte) []byte {
 
 // GetBlock returns Block by the given hash if it exists in the store.
 func (dao *Simple) GetBlock(hash util.Uint256) (*block.Block, error) {
-	key := storage.AppendPrefix(storage.DataBlock, hash.BytesBE())
+	key := storage.AppendPrefix(storage.DataExecutable, hash.BytesBE())
 	b, err := dao.Store.Get(key)
 	if err != nil {
 		return nil, err
 	}
 
-	block, err := block.NewBlockFromTrimmedBytes(dao.Version.StateRootInHeader, b)
+	r := io.NewBinReaderFromBuf(b)
+	if r.ReadB() != storage.ExecBlock {
+		return nil, errors.New("internal DB inconsistency")
```

Comment:
As long as we trust SHA256 this can't happen. Block is very much different from transaction, so we need to have a hash collision and while it's not impossible, it's highly unlikely to happen.